WooCommerce/Classes/Tools/Shared Site Settings/SelectedSiteSettings.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+import Yosemite
+
+/// Shared Settings for the selected Site
+///
+final class SelectedSiteSettings: NSObject {
+    /// Shared Instance
+    ///
+    static let shared: SelectedSiteSettings = {
+        let siteSettings = SelectedSiteSettings()
+        siteSettings.configureResultsController()
+        return siteSettings
+    }()
+
+    /// ResultsController: Whenever settings change, I will change. We both change. The world changes.
+    ///
+    private lazy var resultsController: ResultsController<StorageSiteSetting> = {
+        let storageManager = ServiceLocator.storageManager
+        let descriptor = NSSortDescriptor(keyPath: \StorageSiteSetting.siteID, ascending: false)
+        return ResultsController<StorageSiteSetting>(storageManager: storageManager, sortedBy: [descriptor])
+    }()
+}
+
+// MARK: - ResultsController
+//
+extension SelectedSiteSettings {
+
+    /// Refreshes the currency settings for the current default site
+    ///
+    func refresh() {
+        refreshResultsPredicate()
+    }
+
+    /// Setup: ResultsController
+    ///
+    private func configureResultsController() {
+        resultsController.onDidChangeObject = { (object, indexPath, type, newIndexPath) in
+            CurrencySettings.shared.updateCurrencyOptions(with: object)
+        }
+        refreshResultsPredicate()
+    }
+
+    private func refreshResultsPredicate() {
+        guard let siteID = ServiceLocator.stores.sessionManager.defaultStoreID else {
+            DDLogError("Error: no siteID found when attempting to refresh CurrencySettings results predicate.")
+            return
+        }
+
+        let sitePredicate = NSPredicate(format: "siteID == %lld", siteID)
+        let settingTypePredicate = NSPredicate(format: "settingGroupKey ==[c] %@", SiteSettingGroup.general.rawValue)
+        resultsController.predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [sitePredicate, settingTypePredicate])
+        try? resultsController.performFetch()
+        resultsController.fetchedObjects.forEach {
+            CurrencySettings.shared.updateCurrencyOptions(with: $0)
```

Comment:
If I understand the `ServiceLocator` pattern correctly, shouldn't this access the `CurrencySettings` though the `ServiceLocator` so we can access the mocked/"injected" instance and not the singleton directly?

I wonder if a way to improve this singleton management is to make those `shared` accessors private to `ServiceLocator` and somehow forbid any `shared` usage on the codebase? 😅 

cc @woocommerce/ios-developers

Reply:
Do you mean to remove `CurrencySettings` from being a singleton, and use its instance by adding a new static variable in `ServiceLocator`? If you mean this, I think that it's a good idea to not have an extra singleton, and use it as a global dependency inside the `ServiceLocator`

Reply:
Yes, something like that!

Reply:
Sounds good to me, I'm going to edit the code for including `CurrencySettings` in `ServiceLocator` 👍 I think it make sense

Reply:
Noice! I'll probably get to documenting this in HACK Week. 🤞